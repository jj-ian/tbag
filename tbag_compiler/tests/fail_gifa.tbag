room {}

room Closet { 
	name = "Closet";
}

room Bedroom {
	name = "Bedroom";
}

Closet <-> Bedroom;

start { Closet }

npc {
	string name;
	string roomName;
	int hunger; 
}

npc Cat {
	name = "Tubbs";
	roomName = "Bedroom";
	hunger = 5;
}

true {
	printCurrentRoomInfo();
	getInputAdjacentRooms(Outside);
	->input
}

currentRoom.name ~~ Cat.roomName {
	print("you got eaten by the cat.\n");
	endgame;
}

func void printCurrentRoomInfo() {
	print("Currently in: ");
	print(currentRoom.name);
	print("\n");
}
