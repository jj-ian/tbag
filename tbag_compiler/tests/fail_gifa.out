Fatal error: exception Failure("undeclared identifier Outside")
